### DISCUSSION TITLE
File Uploader using Route Handlers

### DISCUSSION THREAD

**Discussion by leaen** [vercel/next.js#51807] [App Router] [answered]
I am very new to next.js. Is it a good idea to build File Handler using Route Handlers to google drive. I have the user upload 100 images at once to google drive folder. My two major concerns are:

[Payload size limit](https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit): Since uploading 100 images at once size will definitely greater than 4.5mb
[Execution Timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout): Uploading 100 images to google drive will take time > 10 seconds.

To avoid this, I am thinking of deploying a separate express backend.

Am I thinking right? 

Appreciate any suggestions!

**Comment by icyJoseph** [comment, marked as answer]
For those payloads, I think so yeah. The API routes are not made to be a replacement for a CDN, both in terms of uploading and serving.

**Reply by leaen** [reply]
Got it thank you. For image uploading and downloading a separate express server deployed on a cloud provider is the best approach as per your experience?

**Reply by icyJoseph** [reply]
I should make precision, that this is mostly in terms of serverless hosting, such as Vercel. 

If you are hosting it yourself, then there shouldn't be any major issues.

**Reply by icyJoseph** [reply]
To answer your question. Yeah, that's typically better. However, since you are doing this yourself (managing upload and serving) you have to make sure to cache things properly. Having an actual CDN is the best approach.

**Reply by leaen** [reply]
Got it thank you very much for the advice!
